node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -4227,11 +4568,24 @@ mod tests {
         }
     }
 
+    fn node_by_ip_mut(db: &mut NeighborhoodDatabase, ip_addr: IpAddr) -> &mut NodeRecord {
```

Comment:
I still think this might be worth investigating possible reasons of why it cannot in first place be found under the `NeighborhoodDatabase` as a method... If you don't find a significant reason it'd be better having it there rather than here.

Reply:
Good idea.